src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingGraph.java

```diff
@@ -788,6 +788,17 @@ final DanglingChainMergeHelper generateCigarAgainstUpwardsReferencePath(final Mu
         return findPath(vertex, pruneFactor, v -> isReferenceNode(v) || outDegreeOf(v) != 1, v -> isReferenceNode(v), v -> outgoingEdgeOf(v), e -> getEdgeTarget(e));
     }
 
+    /**
+     * Finds a path starting from a given vertex and satisfying various predicates
+     *
+     * @param vertex   the original vertex
+     * @param pruneFactor  the prune factor to use in ignoring chain pieces
+     * @param done predicate on vertices testing whether the end of the path has been reached
```

Comment:
Could this be something like: test for whether a vertex is at the end of the path?